src/pages/cli/graphql/data-modeling.mdx

```diff
@@ -549,7 +549,494 @@ type Address @model {
 ```
 
 ### Relational models with custom primary keys
-TODO
+You can define custom primary keys for relational models. The transformer will handle the foreign key fields needed by queries and mutations, which allows you to retrieve data across different relations. 
+
+You can choose the default relations with **implicit** fields. This will result in auto generation of additional foreign key fields refering its counterpart primary key and sort key fields.
+
+<Callout>
+In "has many" relations, the foreign key fields are only generated in child type.
+</Callout>
+
+<BlockSwitcher>
+
+<Block name='Implicit Bi-directional "has one"'>
```

Comment:
Bidirectional one-to-one could be more precise. Ditto "implicit" term.

Reply:
The `Bi-directional "has one"` is derived from the previous section